extensions/amp/bundle/src/main/java/com/adobe/cq/wcm/core/extensions/amp/internal/AmpModeForwardFilter.java

```diff
@@ -65,57 +66,55 @@ public void doFilter(final ServletRequest request, final ServletResponse respons
 
         SlingHttpServletRequest slingRequest = (SlingHttpServletRequest) request;
 
-        // Read the full selector string from the request.
-        String selectors = slingRequest.getRequestPathInfo().getSelectorString();
-        if (selectors == null) {
-            selectors = "";
-        }
-
-        boolean hasAmpSelector = selectors.contains(AMP_SELECTOR);
+        String[] selectorArray = slingRequest.getRequestPathInfo().getSelectors();
+        boolean hasAmpSelector = ArrayUtils.contains(selectorArray, AMP_SELECTOR);
 
         String ampMode = AmpUtil.getAmpMode(slingRequest);
-
-        if (hasAmpSelector && (ampMode.equals(NO_AMP) || ampMode.isEmpty())) {
-
-            // Remove the amp selector.
-            String newSelectors;
-            if (selectors.contains(DOT + AMP_SELECTOR)) {
-                newSelectors = selectors.replace(DOT + AMP_SELECTOR, "");
-            } else if (selectors.contains(AMP_SELECTOR + DOT)) {
-                newSelectors = selectors.replace(AMP_SELECTOR + DOT, "");
-            } else {
-                newSelectors = selectors.replace(AMP_SELECTOR, "");
-            }
-
-            // Apply the updated selectors.
-            RequestDispatcherOptions options = new RequestDispatcherOptions();
-            options.setReplaceSelectors(newSelectors);
-
-            if (forward(slingRequest, response, options)) {
-                return;
-            }
-        } else if (!hasAmpSelector && ampMode.equals(AMP_ONLY)) {
-
-            // Add the 'amp' selector to the dispatcher. Making sure to put it at the beginning of the selector list.
-            RequestDispatcherOptions options = new RequestDispatcherOptions();
-            if (!selectors.isEmpty()) {
-                options.setReplaceSelectors(AMP_SELECTOR + DOT + selectors);
-            } else {
-                options.setAddSelectors(AMP_SELECTOR);
-            }
-
-            if (forward(slingRequest, response, options)) {
-                return;
-            }
-        } else if (selectors.contains(DOT + AMP_SELECTOR)) {
-
-            // Move the amp selector to the front of the list of selectors.
-            RequestDispatcherOptions options = new RequestDispatcherOptions();
-            String newSelectors = selectors.replace(DOT + AMP_SELECTOR, "");
-            options.setReplaceSelectors(AMP_SELECTOR + DOT + newSelectors);
-
-            if (forward(slingRequest, response, options)) {
-                return;
+        if (hasAmpSelector || ampMode.equals(AMP_ONLY)) {
+
+            String selectors = StringUtils.defaultIfEmpty(slingRequest.getRequestPathInfo().getSelectorString(), StringUtils.EMPTY);
```

Comment:
I would rather we use https://sling.apache.org/apidocs/sling9/org/apache/sling/api/request/RequestPathInfo.html#getSelectors-- to manipulate selectors. This would simplify the logic and avoid future errors caused by string replacement. We already have these available as `selectorArray`